### PR TITLE
Backport missing PRs from Bitcoin 0.14 - Part 9

### PR DESCRIPTION
**Pull request by codablock**
Continuation of https://github.com/dashpay/dash/pull/1770

~We are nearly done with 0.14. After this one I'll create a (hopefully) final PR for 0.14 with https://github.com/bitcoin/bitcoin/pull/8391 (need to figure out the correct BIP heights for Dash) and other PRs I skipped so far.~

UPDATE: I splitted up the PR into 2 as it got too big, so this one is not the last one.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1707,9 +1707,18 @@ UniValue listsinceblock(const JSONRPCRequest& request)
...
+            {
+                // the block being asked for is a part of a deactivated chain;
+                // we don't want to depend on its perceived height in the block
+                // chain, we want to instead use the last common ancestor
+                pindex = chainActive.FindFork(pindex);
+            }
+        }
         else
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid blockhash");
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid blockhash");
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp]: trailing whitespaces

File: src/chainparams.cpp [hunk 2]
```diff
@@ -483,13 +483,17 @@ class CDevNetParams : public CChainParams {
...
+            (      0, uint256S("0x000008ca1832a4baf228eb1553c03d3a2c8e02399550dd6ea8d65cec3ef23d2e"))
+            (      1, devnetGenesis.GetHash())
         };
 
+        chainTxData = ChainTxData{
+            // Data as of block 0000024bc3f4f4cb30d29827c13d921ad77d2c6072e586c7f60d83c2722cdcc5 (height 2999)
+            devnetGenesis.GetBlockTime(),
+            2,     // * total number of transactions between genesis and last checkpoint
+            //   (the tx=... number in the SetBestChain debug.log lines)
+            0.01        // * estimated number of transactions per day after checkpoint
```
Comment (UdjinM6) [inline src/chainparams.cpp]: should say `// * estimated number of transactions per second after that timestamp`

File: src/rpc/client.cpp [hunk 3]
```diff
@@ -136,6 +136,7 @@ static const CRPCConvertParam vRPCConvertParams[] =
     { "prioritisetransaction", 2, "fee_delta" },
     { "setban", 2, "bantime" },
     { "setban", 3, "absolute" },
+    { "setbip69enabled", 0 },
```
Comment (UdjinM6) [inline src/rpc/client.cpp]:
```
rpc/client.cpp:139:28: warning: missing field 'paramName' initializer [-Wmissing-field-initializers]
    { "setbip69enabled", 0 },
                           ^
```

File: src/consensus/params.h [hunk 4]
```diff
@@ -56,15 +56,15 @@ struct Params {
...
     /** Block height and hash at which BIP34 becomes active */
     int BIP34Height;
     uint256 BIP34Hash;
+    /** Block height at which BIP65 becomes active */
+    int BIP65Height;
+    /** Block height at which BIP66 becomes active */
+    int BIP66Height;
     /**
-     * Minimum blocks including miner confirmation of the total of nMinerConfirmationWindow blocks in a retargetting period,
+     * Minimum blocks including miner confirmation of the total of 2016 blocks in a retargetting period,
```
Comment (UdjinM6) [inline src/consensus/params.h]: `nMinerConfirmationWindow` was not a typo :) should s/retargetting/retargeting/ instead

File: src/qt/sendcoinsdialog.cpp [hunk 5]
```diff
@@ -720,8 +720,8 @@ void SendCoinsDialog::updateGlobalFeeVariables()
         // set nMinimumTotalFee to 0 to not accidentally pay a custom fee
         CoinControlDialog::coinControl->nMinimumTotalFee = 0;
 
-        // show the estimated reuquired time for confirmation
-        ui->confirmationTargetLabel->setText(GUIUtil::formatDurationStr(nConfirmTarget * Params().GetConsensus().nPowTargetSpacing)+" / "+tr("%n block(s)", "", nConfirmTarget));
+        // show the estimated required time for confirmation
+        ui->confirmationTargetLabel->setText(GUIUtil::formatDurationStr(nConfirmTarget * Params().GetConsensus().nPowTargetSpacing) + " / "+tr("%n block(s)", "", nConfirmTarget));
```
Comment (UdjinM6) [inline src/qt/sendcoinsdialog.cpp]: should probably do the same to `"+tr`

**Comment by codablock** [comment]
Pushed review fixes. The dash-tx test files that needed Dashification were all SegWit related so I removed them.

I also pushed updates to the BIP34/BIP65/BIP66 heights in chainparams.cpp. I found these heights by running the code in https://github.com/codablock/dash/tree/find_bips_heights (if you want to verify this, run the code until all headers are synced and check the logs). I also verified correctness by running full syncs in mainnet and testnet.

**Comment by UdjinM6** [comment]
👍 , will test.

PS. I still see `src/test/data/txcreateoutpubkey2.*` and `src/test/data/txcreatescript3.*` in files list and in `src/Makefile.test.include`... shouldn't they be dropped as well?

**Comment by codablock** [comment]
Yepp, txcreateoutpubkey2 and txcreateoutpubkey3 shouldn't be in the list and also the files should have been removed. Squashed this into the segwit removal commit and force pushed.

**Comment by UdjinM6** [comment]
ok, `src/test/data/txcreateoutpubkey2.*` is gone now but `src/test/data/txcreatescript3.*` is still here...

**Comment by codablock** [comment]
Argh, sorry about that...I should (and will) go to bed now. Pushed squashed commit again.

**Pull request merged**